thirdeye/thirdeye-frontend/app/pods/services/query-cache/service.js

```diff
@@ -0,0 +1,49 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+/**
+ * @type {Ember.Service}
+ * @summary This service is to also cached the query response data for `this.get('store').query()` calls. In general,
+   the call to `this.get('store').query()` does not do auto cache under the hood.
+   The `onion wrapper` cache only caches the info that ember-data does not currently cache for us. Meaning if the internal records
+   in the ember store changes it changes for this cache as well.
+   This is caching the `onion wrapper` we get back with the data document.
+   An `Ember.Service` is a long-lived Ember object that can be made available in different parts of your application.
+ * @example  a typical json-api doc.
+    {
+      links: {}
+      meta: {},
+      data: [] | {}  <- Note: this will have the entity or entities that the ember-data will cache in its store.
+    }
+ * @see {@link http://jsonapi.org/|jsonapi.org}
+ * //To debug what is in the store - this.get('store')._identityMap.
+ * @example   anomaliesApiService: service('services/api/anomalies'),
+ */
+export default Service.extend({
+  init() {
+    this._super();
+    this._cache = Object.create(null);//create our cache
+  },
+
+  store: service('store'),
+
+  query(type, query, adapterOptions) {
+    assert('you must pass adapterOptions to queryCache.query', adapterOptions);
+    assert('you must pass adapterOptions.cacheKey to queryCache.query', adapterOptions.cacheKey);
+
+    let cached = this._cache[adapterOptions.cacheKey];
+    if (!cached || adapterOptions.reload) {
+       cached = this._cache[adapterOptions.cacheKey] = this.get('store').query(type, query, adapterOptions);//add to our dictionary
+    }
+
+    return cached;
+  },
+  /*
+  We can use the `url` as the `cacheKey` in the `query` hook.
+  which would be different with each change in query params.
+  */
+  urlForQueryKey(modelName, query) {
```

Comment:
let's document the params in the doc above, also, let's stick to the JS doc comment syntax: 
```
/**
 * This is a JS doc comment
 * @type/@method
 * @param
 * @example
 * @return
 */
```